crawlerr/services/rag_service.py

```
@@ -73,6 +128,228 @@ def chunk_text(self, text: str, metadata: Dict[str, Any] = None) -> List[Dict[st
+        offsets = []
+        
+        if self.tokenizer_type == "qwen":
+            # For HF tokenizers, use built-in offset mapping when available
+            try:
+                encoding = self.tokenizer(
+                    text, 
+                    return_offsets_mapping=True, 
+                    add_special_tokens=False
+                )
+                return encoding["offset_mapping"]
+            except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)